backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -194,4 +208,63 @@ const IR::Node* RemoveUnusedMetadataFields::preorder(IR::DpdkAsmProgram *p) {
     return p;
 }
 
+bool ValidateTableKeys::isMetadataStruct(const IR::Type_Struct *st) {
+       for (auto anno : st->annotations->annotations) {
+            if (anno->name == "__metadata__") {
+                return true;
+            }
+       }
+       return false;
+}
+
+bool ValidateTableKeys::preorder(const IR::DpdkAsmProgram *p) {
+    const IR::DpdkStructType *metaStruct;
+    for (auto st : p->structType) {
+        if (isMetadataStruct(st)) {
+            metaStruct = st;
```

Comment:
Are there multiple metadata struct possible? Is it necessary to overwrite metaStruct? Can we break after first find?

Reply:
updated.